packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6194,6 +6212,10 @@ class MultiChildRenderObjectElement extends RenderObjectElement {
     super.update(newWidget);
     assert(widget == newWidget);
     _children = updateChildren(_children, widget.children, forgottenChildren: _forgottenChildren);
+    assert(() {
+      _children.forEach(_debugCheckHasAssociatedRenderObject);
```

Comment:
If we check after `updateChildren`, some scenarios where keys are reused will throw more exceptions. See the test cases for details.
If we want to avoid this happens, we need to extend the `updateChildren` parameter and pass the check function to it. However, if we do this, the modification is very invasive. What is your opinion?

Reply:
@chunhtai

Reply:
It seems like that we can override the inflateWidge and make the check instead. I will try it.

Reply:
@chunhtai Hi,
If we make checking after `updateChildren` like this commit, it will break some tests because retaken the globalKey will re-parent the RO and trigger the new assert first. Also, need to traverse all the children extra.

Reply:
And we are not easy to rewrite the `Element.updateChildren `, because it is too complicated.